src/Poseidon/GenotypeData.hs

```diff
@@ -256,18 +259,30 @@ recodeAlleles consensusSnpEntry snpEntry genoLine = do
     flipGeno HomAlt = HomRef
     flipGeno g      = g
 
-printSNPCopyProgress :: (MonadIO m) => Pipe a a m ()
-printSNPCopyProgress = do
+printSNPCopyProgress :: (MonadIO m) => LogEnv -> UTCTime -> Pipe a a m ()
+printSNPCopyProgress logEnv startTime = do
     counterRef <- liftIO $ newIORef (0 :: Int)
     for cat $ \val -> do
         n <- liftIO $ readIORef counterRef
-        when (n `rem` 1000 == 0) $ do
-            liftIO $ hClearLine stderr
-            liftIO $ hSetCursorColumn stderr 0
-            liftIO $ hPutStr stderr ("> " ++ show n ++ " ")
-            liftIO $ hFlush stderr
+        currentTime <- liftIO getCurrentTime
+        logWithEnv logEnv $ logProgress n (diffUTCTime currentTime startTime)
         liftIO $ modifyIORef counterRef (+1)
         yield val
+    where
+        logProgress :: Int -> NominalDiffTime -> PoseidonLogIO ()
+        logProgress c t
+            | c >=  1000000 &&                c `rem` 1000000 == 0 = logInfo $ "SNPs: " ++ pad (show c) ++ " - " ++ prettyTime t
```

Comment:
Very cool idea to stack it with order of magnitude! I will have to try that out myself, but code-wise definitely looks very cool!

Reply:
I read that one should avoid slowing the progress indication down. For psychological reasons. So I'm not sure if this is really the best solution. Could we easily (and quickly) determine the number to SNPs to be merged in a given forge operation? Then we could do a linearly scaled progress update. :thinking:

Reply:
Hmm... I suppose quickly counting the lines of the bim file would do the job. I just tried with `wc -l` on the command line and with 1240K data it goes pretty much instantaneous. I think using my bim-reader from `sequence-formats` will be slower, since it also parses it into Haskell structures. Perhaps there is a quick line-counting routine somewhere in the Haskell world? The turtle library might have a function for that... but again, whether it's as fast as `wc -l` I don't know.

Reply:
To discuss this at one place:
> I think the non-linear printing of SNP conversion is indeed a bit confusing. I think I would try and find a linear logging with the assumption that 99% of use cases will have around 1,2 million SNPs... so I think printing a line every 10,000 SNPs seems acceptable to me. If people forge complete-genome data with 60M SNPs that means 6000 log-lines, which is the maximum and still seems not too terrible to me.

6000 lines is ridiculous, but how often will somebody merge that many SNPs? Maybe the simplest solution is really the best. 10000 doesn't sound too bad.

Reply:
I decided to implement this now. It's verbose, but it is 100% clear and reliable.